scripts/build.py

```diff
@@ -609,6 +611,80 @@ def build_files(source, target, file_hashes):
             print e
 
 
+def build_HTML_files(source, target, file_hashes):
+    """Minifies HTML files and interpolates paths in HTML to include hashes in
+    source directory and copies it to target.
+    Args:
+        source: str. Path relative to /oppia directory of directory
+            containing files and directories to be copied.
+        target: str. Path relative to /oppia directory of directory where
+            to copy the files and directories.
+        file_hashes: dict(str, str). Dictionary of file hashes.
+    """
+    print 'Processing %s' % os.path.join(os.getcwd(), source)
+    print 'Generating into %s' % os.path.join(os.getcwd(), target)
+    ensure_directory_exists(target)
+    for root, dirs, files in os.walk(os.path.join(os.getcwd(), source)):
+        for directory in dirs:
+            print 'Building %s' % os.path.join(root, directory)
+        for filename in files:
+            if filename.endswith('.html'):
+                source_path = os.path.join(root, filename)
+                if target in source_path:
+                    continue
+                if source not in source_path:
+                    continue
+                target_path = source_path.replace(source, target)
+                ensure_directory_exists(target_path)
+                process_html(source_path, target_path, file_hashes)
+
+
+def rebuild_new_files(
+        source_path, target_path, recently_changed_filenames, file_hashes):
+    """Minify list of recently changed files.
+
+    Args:
+        source_path: str. Path relative to /oppia directory of directory
+            containing files and directories to be copied.
+        target_path: str. Path relative to /oppia directory of directory where
+            to copy the files and directories.
+        recently_changed_filenames: list(str). List of recently changed files.
+        file_hashes: dict(str, str). Dictionary of file hashes.
+    """
+    for file_name in recently_changed_filenames:
+        print 'Minifying %s' % file_name
+        source_file_path = os.path.join(source_path, file_name)
+        target_file_path = os.path.join(target_path, file_name)
+        ensure_directory_exists(target_file_path)
+        minify_func(source_file_path, target_file_path, file_hashes, file_name)
+
+
+def get_new_files_from_directory(source, target):
+    """Compare current's hashed file and newly generated hash and return list of
+    expected new files.
+
+    Args:
+        source: str. Path relative to /oppia where the DEV files are located.
+        target: str. Path relative to /oppia where the BUILD files are located.
+
+    Returns:
+        array(str). List of files expected to be re-hashed.
```

Comment:
Here and elsewhere: distinguish between filenames and files. I think you're returning an array of filenames. Also the python type is list, not array.

Reply:
Updated docstring for incorrect return type